src/main/java/io/github/moulberry/notenoughupdates/core/util/Vec3Comparable.java

```diff
@@ -0,0 +1,124 @@
+package io.github.moulberry.notenoughupdates.core.util;
+
+import net.minecraft.util.Vec3;
+import net.minecraft.util.Vec3i;
+
+public class Vec3Comparable extends Vec3 implements Comparable<Vec3Comparable> {
+	public static final Vec3Comparable NULL_VECTOR = new Vec3Comparable(0.0, 0.0, 0.0);
+
+	public Vec3Comparable(double x, double y, double z) {
+		super(x, y, z);
+	}
+
+	public Vec3Comparable(Vec3i sourceVec) {
+		super(sourceVec);
+	}
+
+	public Vec3Comparable(Vec3 source) {
+		super(source.xCoord, source.yCoord, source.zCoord);
+	}
+
+	public Vec3Comparable(Vec3Comparable source) {
+		super(source.xCoord, source.yCoord, source.zCoord);
+	}
+
+	@Override
+	public Vec3Comparable subtractReverse(Vec3 vec) {
+		return new Vec3Comparable(super.subtractReverse(vec));
+	}
+
+	@Override
+	public Vec3Comparable normalize() {
+		return new Vec3Comparable(super.normalize());
+	}
+
+	@Override
+	public Vec3Comparable crossProduct(Vec3 vec) {
+		return new Vec3Comparable(super.crossProduct(vec));
+	}
+
+	@Override
+	public Vec3Comparable subtract(Vec3 vec) {
+		return new Vec3Comparable(super.subtract(vec));
+	}
+
+	@Override
+	public Vec3Comparable subtract(double x, double y, double z) {
+		return new Vec3Comparable(super.subtract(x, y, z));
+	}
+
+	@Override
+	public Vec3Comparable add(Vec3 other)	{
+		return new Vec3Comparable(super.add(other));
+	}
+
+	@Override
+	public Vec3Comparable addVector(double x, double y, double z) {
+		return new Vec3Comparable(super.addVector(x, y, z));
+	}
+
+	@Override
+	public Vec3Comparable getIntermediateWithXValue(Vec3 vec, double x) {
+		return new Vec3Comparable(super.getIntermediateWithXValue(vec, x));
+	}
+
+	@Override
+	public Vec3Comparable getIntermediateWithYValue(Vec3 vec, double y) {
+		return new Vec3Comparable(super.getIntermediateWithYValue(vec, y));
+	}
+
+	@Override
+	public Vec3Comparable getIntermediateWithZValue(Vec3 vec, double z) {
+		return new Vec3Comparable(super.getIntermediateWithZValue(vec, z));
+	}
+
+	@Override
+	public Vec3Comparable rotatePitch(float pitch) {
+		return new Vec3Comparable(super.rotatePitch(pitch));
+	}
+
+	@Override
+	public Vec3Comparable rotateYaw(float yaw) {
+		return new Vec3Comparable(super.rotateYaw(yaw));
+	}
+
+	@Override
+	public boolean equals(Object other)	{
```

Comment:
This method technically breaks the symmetry contract of equals: 
```java
var a = new Vec3(0,0,0);
var b = new Vec3Comparable(0,0,0);
a.equals(b) // false
b.equals(a) // true```

Reply:
Good catch - I'll fix it.